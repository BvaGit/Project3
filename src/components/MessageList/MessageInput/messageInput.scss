@import "../../../utils/variables.scss";

.message {
  position: absolute;
  bottom: 40px;
  width: 100%;
  height: 70px;
  align-items: center;
  margin-bottom: 10px;

  &__input {
    background-color: #d6d6d6;
    width: 700px;
    border: 2px solid $background;
    border-radius: 5px;
    height: 80px;
    padding: 5px 10px;
    color: $title-color;

    &::placeholder {
      color: $title-color;
    }

    &::-webkit-input-placeholder {
      opacity: 1;
      transition: opacity 0.3s ease;
    }
    &::-moz-placeholder {
      opacity: 1;
      transition: opacity 0.3s ease;
    }
    &:-moz-placeholder {
      opacity: 1;
      transition: opacity 0.3s ease;
    }
    &:-ms-input-placeholder {
      opacity: 1;
      transition: opacity 0.3s ease;
    }
    &:focus::-webkit-input-placeholder {
      opacity: 0;
      transition: opacity 0.3s ease;
    }
    &:focus::-moz-placeholder {
      opacity: 0;
      transition: opacity 0.3s ease;
    }
    &:focus:-moz-placeholder {
      opacity: 0;
      transition: opacity 0.3s ease;
    }
    &:focus:-ms-input-placeholder {
      opacity: 0;
      transition: opacity 0.3s ease;
    }
  }

  &__btn {
    width: 0;
    height: 0;
    border-top: 30px solid transparent;
    border-left: 80px solid $cross;
    border-bottom: 30px solid transparent;
    margin-left: 20px;

    &:hover {
      border-left: 80px solid #008894;
    }

    &:active {
      border-left: 80px solid #7d8c8d;
    }
  }
}

.theme-dark {
  .message__input {
    background-color: $l-body-background;
    color: $background;

    &::placeholder {
      color: #fff;
    }
  }
}
