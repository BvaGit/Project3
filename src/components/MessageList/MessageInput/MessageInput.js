import React, { useEffect, useState } from "react";

import { getTime } from "../../../helpers/message";

import "./messageInput.scss";

const MessageInput = ({ socket, getUsersFromApi, userId, chatId, rooms }) => {
  const [content, setContent] = useState("");

  useEffect(() => {
    getUsersFromApi();
  }, [userId]);

  const handleChangeInput = (e) => {
    setContent(e.target.value);
  };

  const sendMessage = () => {
    if (content) {
      socket.emit("send_message", {
        id: userId,
        chat_id: chatId,
        content: content,
        date_create: getTime(),
      });
      setContent("");
    }
  };

  return (
    <>
      <div className="message">
        <label>
          <input
            key={rooms.chat_id}
            value={content}
            className="message__input"
            onChange={handleChangeInput}
            placeholder="Write a message"
          />
        </label>
        <button
          className="message__btn"
          onClick={sendMessage}
          type="submit"
        ></button>
      </div>
    </>
  );
};

export default MessageInput;
