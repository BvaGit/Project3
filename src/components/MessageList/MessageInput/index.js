import MessageInput from "./MessageInput";
import { connect } from "react-redux";

import { getSocket } from "../../../store/rooms/selectors";
import { sendMessages } from "../../../store/rooms/asyncActions";
import { getUsersFromApi } from "../../../store/users/asyncActions";
import { getUsersFromStore } from "../../../store/users/selectors";
import { getUserChats } from "../../../store/rooms/asyncActions";
import { setActiveRoom } from "../../../store/rooms/actions";
import { getUserID } from "../../../store/user/selectors";
import { getRooms } from "../../../store/rooms/selectors";

const mapStateToProps = (state) => ({
  socket: getSocket(state),
  users: getUsersFromStore(state),
  userId: getUserID(state),
  rooms: getRooms(state),
});

const mapDispatchToProps = (dispatch) => ({
  sendMessages: (payload) => dispatch(sendMessages(payload)),
  getUsersFromApi: (payload) => dispatch(getUsersFromApi(payload)),
  getUserChats: (userId) => dispatch(getUserChats(userId)),
  setActiveRoom: (chatId) => dispatch(setActiveRoom(chatId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(MessageInput);
