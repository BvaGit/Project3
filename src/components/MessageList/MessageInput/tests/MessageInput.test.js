import React from 'react';
import { shallow, mount } from 'enzyme';
import MessageInput from '../MessageInput';


describe('MessageInput', () => {
    let props;
    it("Should match snapshot", () => {
        props={rooms:{chat_id:1},socket:{emit:jest.fn()},userId:1, getUsersFromApi:jest.fn()}
        const component = shallow(<MessageInput {...props}/>)
        expect(component.html()).toMatchSnapshot()
    })
    it("Should change input", () => {
        props={rooms:{chat_id:1},socket:{emit:jest.fn()}, userId:1, getUsersFromApi:jest.fn()}
        const component = mount(<MessageInput {...props}/>)
        component.find('input').simulate("change", {target:{value:"value"}})
        expect(component.find('input').value).toBe()
    })
    it("Should click button", () => {
        props={rooms:{chat_id:1},socket:{emit:jest.fn()}, userId:1, getUsersFromApi:jest.fn()}
        const component = mount(<MessageInput {...props}/>)
        component.find('button').simulate('click')
        component.find('input').simulate("change", {target:{value:"value"}})
        component.find('button').simulate('click')
        expect(props.socket.emit).toHaveBeenCalled()
    })
})
 