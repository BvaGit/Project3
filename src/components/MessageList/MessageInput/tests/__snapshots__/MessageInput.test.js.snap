// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput Should match snapshot 1`] = `"<div class=\\"message\\"><label><input value=\\"\\" class=\\"message__input\\" placeholder=\\"Write a message\\"/></label><button class=\\"message__btn\\" type=\\"submit\\"></button></div>"`;
